test/functional/tests/io_class/test_io_classification.py

```diff
@@ -70,37 +70,40 @@ def test_ioclass_lba():
                     .count(dd_count)
                     .block_size(dd_size)
                     .seek(lba)
+                    .oflag("direct")
             )
             dd.run()
-            sync()
             dirty_count += 1
 
             dirty = cache.get_io_class_statistics(io_class_id=ioclass_id).usage_stats.dirty
             if dirty.get_value(Unit.Blocks4096) != dirty_count:
                 TestRun.LOGGER.error(f"LBA {lba} not cached")
 
-    with TestRun.step("Flush cache."):
-        cache.flush_cache()
-
     with TestRun.step("Run IO and check if lba outside of defined range are not cached."):
-        TestRun.LOGGER.info(f"Writing to random sectors outside of cached range.")
-        for i in range(iterations):
-            rand_lba = random.randrange(2000)
-            if min_cached_lba <= rand_lba <= max_cached_lba:
-                continue
+        TestRun.LOGGER.info(f"Writing to sectors outside of cached range.")
+        test_lba = [max_cached_lba + 1] + random.sample(
+            [
+                *range(0, min_cached_lba),
+                *range(max_cached_lba + 1, int(core.size.get_value(Unit.Blocks512)))
```

Comment:
shouldn't this take into account disk block size?

Reply:
I think it shouldn't, actually. I think that Linux block device addressing always is done in 512B blocks, so this should be fine.